### DISCUSSION TITLE
[PnP] Check if dependency exists

### DISCUSSION THREAD

**Discussion by rtritto** [yarnpkg/berry#6323] [Q&A] [answered]
How can I change the check existence (`nodeModulesPath` and `fs.pathExists`) of a dependency?

`dir` = project folder path
`packageName` = [electron](https://www.npmjs.com/package/electron)

https://github.com/electron/forge/blob/38f64e4c9e2c79a3b9ff8447b1a23199415ede97/packages/utils/core-utils/src/electron-version.ts#L37-L43

```ts
 async function determineNodeModulesPath(dir: string, packageName: string): Promise<string | undefined> { 
   const nodeModulesPath: string | undefined = path.join(dir, 'node_modules', packageName); 
   if (await fs.pathExists(nodeModulesPath)) { 
     return nodeModulesPath; 
   } 
   return findAncestorNodeModulesPath(dir, packageName); 
 } 
```

### Current
```ts
fs.pathExists(nodeModulesPath) = false
```

### Expected
```ts
fs.pathExists(nodeModulesPath) = true
```

**Comment by merceyz** [comment, marked as answer]
Use [`require.resolve`](https://nodejs.org/docs/latest-v22.x/api/modules.html#requireresolverequest-options) to locate dependencies regardless of whether PnP is used or not.
```js
require.resolve(`${packageName}/package.json`, { paths: [dir] });
```
